spec.html

```diff
@@ -8555,13 +8555,15 @@ <h1>[[Construct]] ( _argumentsList_, _newTarget_ )</h1>
     </emu-clause>
 
     <emu-clause id="sec-ordinaryfunctioncreate" aoid="OrdinaryFunctionCreate" oldids="sec-functionallocate,sec-functioninitialize,sec-functioncreate,sec-generatorfunctioncreate,sec-asyncgeneratorfunctioncreate,sec-async-functions-abstract-operations-async-function-create">
-      <h1>OrdinaryFunctionCreate ( _functionPrototype_, _ParameterList_, _Body_, _thisMode_, _Scope_ )</h1>
-      <p>The abstract operation OrdinaryFunctionCreate requires the arguments: an object _functionPrototype_, a parameter list Parse Node specified by _ParameterList_, a body Parse Node specified by _Body_, _thisMode_ which is either ~lexical-this~ or ~non-lexical-this~, and a Lexical Environment specified by _Scope_. OrdinaryFunctionCreate performs the following steps:</p>
+      <h1>OrdinaryFunctionCreate ( _functionPrototype_, _definition_, _ParameterList_, _Body_, _thisMode_, _Scope_ )</h1>
```

Comment:
it seems like this should be added last as an optional parameter, then it can default to `~empty~`?

Reply:
https://github.com/tc39/ecma262/issues/1824#issuecomment-581997120

The parameter is almost always provided. I don't think using an optional parameter is worth it.

Reply:
Then, optional or not, why not at the end?

Reply:
The current parameter list has two Parse Node parameters (`_ParameterList_` and `_Body_`) for syntactic components of the function declaration. Since the new parameter is a Parse Node representing the whole declaration, it makes sense to put it next to those existing Parse Node parameters. Putting it before or after is somewhat debatable, but I put the new one before the existing ones because then the order of nonterminals in the invocations of `OrdinaryFunctionCreate` is the same as in the associated production. E.g. for production:

    FunctionDeclaration : `function` `(` FormalParameters `)` `{` FunctionBody `}`

the invocation is:

    OrdinaryFunctionCreate(..., |FunctionDeclaration|, |FormalParameters|, |FunctionBody|, ...)

I think this makes it easier to understand/remember the order of parameters for this operation.